datapipelines/src/test/java/io/customer/datapipelines/DeviceAttributesTests.kt

```diff
@@ -0,0 +1,126 @@
+package io.customer.datapipelines
+
+import android.Manifest
+import android.content.pm.PackageManager
+import com.segment.analytics.kotlin.android.plugins.AndroidContextPlugin
+import io.customer.datapipelines.support.core.RobolectricTest
+import io.customer.datapipelines.support.core.TestConfiguration
+import io.customer.datapipelines.support.core.testConfiguration
+import io.customer.datapipelines.support.extensions.deviceToken
+import io.customer.datapipelines.support.extensions.encodeToJsonValue
+import io.customer.datapipelines.support.utils.OutputReaderPlugin
+import io.customer.datapipelines.support.utils.TestConstants
+import io.customer.datapipelines.support.utils.trackEvents
+import io.customer.sdk.extensions.random
+import io.mockk.every
+import kotlinx.serialization.json.booleanOrNull
+import kotlinx.serialization.json.intOrNull
+import kotlinx.serialization.json.jsonPrimitive
+import org.amshove.kluent.shouldBeEmpty
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldContain
+import org.amshove.kluent.shouldHaveSingleItem
+import org.amshove.kluent.shouldHaveSize
+import org.amshove.kluent.shouldNotBeNull
+import org.amshove.kluent.shouldNotBeNullOrBlank
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE)
+class DeviceAttributesTests : RobolectricTest() {
```

Comment:
i think one test missing would be having both custom attribute and these built-in device attribute and make sure we get all of them.

Reply:
Good call, added test for it